src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -244,16 +245,23 @@ public void setConfig(File file) {
     // -@cs[AbbreviationAsWordInName] Should be removed at 7.0 version,
     // we keep for some time to avoid braking compatibility.
     @Deprecated
-    public void setConfigURL(URL url) {
+    public void setConfigURL(String url) {
         setConfigUrl(url);
     }
 
     /**
      * Sets URL to the configuration.
      * @param url the URL of the configuration to use
      */
-    public void setConfigUrl(URL url) {
-        setConfigLocation(url.toExternalForm());
+    public void setConfigUrl(String url) {
+        try {
+            setConfigLocation(new URL(url).toExternalForm());
```

Comment:
no need to do instantiation of URL here  `final URI uri = CommonUtils.getUriByFilename(config);` , string could stored as is , to be processed at
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java#L179

conversion to URL/URI - https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java#L372